docs/src/_includes/layouts/doc.html

```diff
@@ -148,4 +148,27 @@ <h1>{{ title }}</h1>
     </a>
 </div>
 
+<div id="copy-announcement" class="visually-hidden" aria-live="polite" aria-atomic="true"></div>
+
+<script>
+    (function() {
+        const copyBtns = document.querySelectorAll("#copyBtn");
+        const announcementRegion = document.getElementById("copy-announcement");
+
+        for (const copyBtn of copyBtns) {
+            const copyText = copyBtn.previousElementSibling.firstChild.cloneNode(true).textContent;
+
+            copyBtn.onclick = () => {
+                navigator.clipboard.writeText(copyText);
+                announcementRegion.innerHTML = "Copied!";
+                copyBtn.setAttribute("data-copied", 'true');
+
+                setTimeout(() => {
+                    copyBtn.removeAttribute("data-copied");
+                    announcementRegion.innerHTML = "";
+                }, 2000);
+            }
```

Comment:
```suggestion
                navigator.clipboard.writeText(copyText)
                .then(() => {
                    announcementRegion.innerHTML = "Copied!";
                    copyBtn.setAttribute("data-copied", 'true');
                    setTimeout(() => {
                        copyBtn.removeAttribute("data-copied");
                        announcementRegion.innerHTML = "";
                    }, 2000);
                })
                .catch(err => {
                    announcementRegion.innerHTML = "Failed to copy!";
                    console.error("Clipboard write failed: ", err);
                });
```
We need to use  .then() with error handling when working with the navigator clipboard API because clipboard operations are asynchronous and can fail for several reasons like permission issues and  Clipboard API Access Restrictions.